fuzzer/src/main.rs

```diff
@@ -84,27 +85,66 @@ const TEST_COUNT: usize = 5;
 /// 0 / 1 / 2 / 3
 const DEBUG_LEVEL: u8 = 0;
 
+#[derive(Debug, Clone, Default)]
+struct Pattern {
+    prefix: String,
+    repeating_pattern: String,
+    postfix: String,
+}
+
+impl<'a> Into<Pattern> for &'a str {
+    fn into(self) -> Pattern {
+        Pattern {
+            prefix: String::new(),
+            repeating_pattern: self.to_owned(),
+            postfix: String::new(),
+        }
+    }
+}
+
+struct UniformPatterns<'a> {
+    patterns: &'a [Vec<u8>],
+}
+
+impl<'a> UniformPatterns<'a> {
+    fn new(patterns: &'a [Vec<u8>]) -> Self {
+        Self { patterns }
+    }
+}
+
+impl<'a> Distribution<Pattern> for UniformPatterns<'a> {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Pattern {
+        fn random_seq<'b, I: Iterator<Item = Option<&'b Vec<u8>>>>(iter: &mut I) -> String {
+            String::from_utf8(
+                iter.take(COMBINATIONS)
+                    .flatten()
+                    .flatten()
+                    .cloned()
+                    .collect(),
+            )
+            .expect("Bytes to String failed!")
```

Comment:
Good point, but that would indicate a bug in the fuzzer, right? Under our current assumptions, it *should* never panic. It may be better to fail and abort so that we can learn?

Reply:
I was very torn on the exact same thing. I opted for more safety than security here, because when I wrote this, the automatic restart mechanism wasn't in place yet. With the restart algorithm I don't see the need to be so strict about it anymore.